check_source.py

```diff
@@ -114,8 +148,8 @@ def check_source_submission(self, source_project, source_package, source_revisio
             self.review_messages['declined'] = 'Only one action per request allowed'
             return False
 
-        if source_revision is None:
-            self.review_messages['declined'] = 'Submission not from a pinned source revision'
+        # The SR itself can use the commit number safely, because it's declined on package deletion
```

Comment:
but after recreating the package, the SR can be 'reopened'

Reply:
IIRC I tried this once but got `is in a final state and can't be changed`

Reply:
If we have to enforce hashes in SRs, then pretty much every SR to `oS:F` currently will be declined...

Reply:
Reopening should also restart the reviews. But if it happens quickly enough, then the caches might make this a false positive.

(This rabbit hole goes really deep...)

Reply:
https://build.opensuse.org/request/show/897561

Reply:
declining non-hash SRs can't be enforced without the backend transparently changing the rev - a simple `osc sr DEVELPRJ PKG openSUSE:Factory` creates an sr with a numeric rev

Reply:
For a transition period, the bot could decline the sr and create a new one with hash instead

Reply:
giving the users a decline message on everything and then not having the users as creator makes it hard to enter into dialogs through the SR UI

Should rather be the backend rewriting it from 'rev x' to hash-rev (it has all the information after all)

Reply:
The above SR just proves that we shouldn't try to enforce this client side. This really should be enforced on the source revision system

Reply:
Yes, so far there's no progress or even acknowledgement on the OBS side.

Reply:
Yeah, let me use my levers... :)